### PR TITLE
[system-command-line] instantiate command error handling improvements

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
#3809 
error handling for alias assignment
invalid template logic (duplicate parameters etc)

### Solution
Implemented error handling for alias assignment.

For possible errors in template grouping: they cannot happen as parameter uniqueness etc is validated when template.json is read. So kept exceptions as is without extra handlinsg.

### Checks:
- [x] Added unit tests
- [x] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx [hunk 1]
```diff
@@ -686,4 +686,21 @@ Run 'dotnet {1} --show-aliases' with no args to show all aliases.</value>
     <comment>0} - option name (as --language)
 {1} - comma separated, escaped in quotes list of allowed values</comment>
   </data>
-</root>
+  <data name="AliasAssignmentCoordinator_Error_LongAlias" xml:space="preserve">
+    <value>Failed to assign long option alias for parameter {0}, tried: '{1}'; '{2}'.</value>
+    <comment>{0} is parameter name, {1} and {2} are attempted option aliases for the parameter</comment>
```
Comment (bekir-ozturk) [inline src/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx:691]: I think people translating these may not know what an "attempted option alias" is. Would it be better to just let them know what kind of value could be assigned to these (name, number, path, date-time etc.)? Like:

```
{0} is a parameter name. {0}, {1} and {2} are all single words with no symbols.
```

File: src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs [hunk 2]
```diff
@@ -220,7 +223,16 @@ private void AddTemplateOptionsToCommand(CliTemplateInfo templateInfo)
...
+            IReadOnlyDictionary<CliTemplateParameter, IReadOnlyList<string>> errors = parametersWithAliasAssignments
+                .Where(p => p.Errors.Any())
+                .ToDictionary(p => p.Parameter, p => p.Errors);
+
+            if (errors.Any())
+            {
+                throw new InvalidTemplateParametersException(templateInfo, errors);
+            }
+
+            foreach ((CliTemplateParameter parameter, IReadOnlyList<string> aliases, IReadOnlyList<string> _) in parametersWithAliasAssignments)
```
Comment (bekir-ozturk) [inline src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs:235]: It seems that we don't need the dictionary unless there is an error.
```suggestion
            IReadOnlyDictionary<CliTemplateParameter, IReadOnlyList<string>> errors = parametersWithAliasAssignments
                .Where(p => p.Errors.Any());

            if (errors.Any())
            {
                throw new InvalidTemplateParametersException(templateInfo, errors.ToDictionary(p => p.Parameter, p => p.Errors));
            }

            foreach ((CliTemplateParameter parameter, IReadOnlyList<string> aliases, IReadOnlyList<string> _) in parametersWithAliasAssignments)
```

**Review by bekir-ozturk — APPROVED** [review]

File: src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs (hunk 2 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs:235]: Agreed, refactored.

File: src/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx (hunk 1 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx:691]: It is difficult to come with restrictions list here, I put "no whitespace" but in theory even whitespace will be allowed (as it is allowed in json). We should do better validations of jsons in first place, then we could come up with better validation.

**Pull request merged**